### ISSUE TITLE
Groups in box plot slides out of their frame

### ISSUE THREAD

**Issue by KerenHagitStern** [endangeredoxen/fivecentplots#47]
Hi,
I noticed that when my group headers don't fit the total plot size, they slide out the frames.
Is there a way that the package can fit the plot size/the headers fonts/expend the headers frame, automatically (without an input ) ,so the group headers will be separated well?
I attached some examples to illustrate the problem. 
thanks
![example_1](https://user-images.githubusercontent.com/107401907/187708011-e79ae97b-790a-4215-9eb7-853316e6422b.jpg)
![example_2](https://user-images.githubusercontent.com/107401907/187708025-7fd79898-e55e-45bb-ab59-db8a6db9a9b6.jpg)
![example_3](https://user-images.githubusercontent.com/107401907/187708034-e2604bce-ea85-42c7-966d-fa080439a0b7.jpg)
![example_4](https://user-images.githubusercontent.com/107401907/187708036-c1e40576-7b32-4a6d-bf41-0e232dd03dbf.jpg)

**Comment by endangeredoxen** [comment]
I love this example data set.  

The vertical label size should auto-expand to fit the long group names so this might be a bug (will check).  I purposely avoided trying to auto-scale the font size in cases like this and hadn't allowed wide labels to expand the width to honor the global setting for `ax_size`.  Perhaps a good compromise is to auto-expand the width unless the user explicitly defines the image size.  I will think about it and get back to you

**Comment by endangeredoxen** [comment]
I did some investigation here and tried to mock up your data frame.  What version are you using (`fcp.__version__`)?  I suspect v0.4.3?  On the latest beta release of v0.5.0 I get the following much cleaner result:

![image](https://user-images.githubusercontent.com/14165793/187838261-86e6be05-014a-48bb-aeb9-5b85aaf655b2.png)

There is still an issue with the labels for "number" but this could be fixed by increasing the width in `ax_size`.  I'll look into making this automatically widen but in the meantime, I recommend switching to v0.5.0 for better results

**Comment by endangeredoxen** [comment]
Disregard my message from 15 min ago.  Still investigating

**Comment by endangeredoxen** [comment]
Just a quick update, you have indeed found a bug.  I am working to fix it and will keep you posted.

**Comment by endangeredoxen** [comment]
OK, I have a bug fix and a new feature for you using `pip install fivecentplots==0.5.0b3`

By default, fcp boxplots will continue to use a default axes size of `[400, 400]`, giving you something like this (notice the vertical label bug is gone, but the first row of labels is getting squished):
![default](https://user-images.githubusercontent.com/14165793/191168645-a5823a45-075b-4d37-9e28-378e58f52379.png)

You still have the option to manually set the `ax_size` to deal with the labels not fitting horizontally (here I used `[700, 400]`):
![manual](https://user-images.githubusercontent.com/14165793/191168837-7e9556f9-6120-496d-8b31-3eadb3fe3c80.png)

Now I give you the option to use `ax_size='auto'` which attempts to automatically resize `ax_size[0]` to fit the contents (note, I still consider this "experimental" so your mileage may vary):
![auto](https://user-images.githubusercontent.com/14165793/191168992-970ada96-3baa-4b99-8f23-98adea7296db.png)

Let me know if you find any issues with this.  Hope this gives you what you need.

**Comment by KerenHagitStern** [comment]
Thanks a lot!
I will try it

**Comment by endangeredoxen** [comment]
Great.  Also note that you no longer need to install the beta version to try this as I've released v0.5.0.  So a regular `pip install fivecentplots` is enough

**Comment by KerenHagitStern** [comment]
![Capture](https://user-images.githubusercontent.com/107401907/194257946-39af151f-cda3-4401-85c9-e8ff5f73893b.PNG)
Hi, I tried the auto option and it worked. Can it plot the graph in a way that the X-axis is shortened to a minimum size? for example, if I changed the order of the categories on the x-axis, the ChickenId labels will be printed vertically and not horizontally ( and I prefer they will be printed vertically).
thanks
